### ISSUE TITLE
MD006 / MD007 start_indent bug

### ISSUE THREAD

**Issue by TheJaredWilcurt** [DavidAnson/markdownlint#2234]
Trying to research *why* `MD006` was deprecated and not replaced with the same functionality elsewhere. It looks like theoretically `MD007` could do the same thing, but if that's the intention, it has a bug, and is very unintuitive.

Goal:

Given this file:

```md
    * This file has one bullet point preceeded by 4 spaces
```

The linter should throw an error that the first level should not be indented.

Previously you would just use:

```yml
# MD006 - Consider starting bulleted lists at the beginning of the line
ul-start-left: true
```

Easy peasy. Or it was, now it's been removed. And I've been using a 6 year old version of the library ever since to retain this functionality. Trying to upgrade now, and running into this.

Theoretically I should just be able to do:

```yml
# MD007 - Unordered list indentation
ul-indent:
  start_indented: false
```

And that would logically prevent unordered lists from starting with indentation. But a-ha! it doesn't! Because, unintuitively, it only applies `start_indented` if there is also a `start_indent` value set. Which is totally backwards from what I'm trying to do (and in fact, this used to be documented as the *reverse* of MD006 since MD007 are both solving opposite intentions).

Okay... So lets just set all 3 options for this rule:

```yml
ul-indent:
  indent: 4
  start_indent: 0
  start_indented: true
```

There we go, top level UL's should have an indentation of 0 spaces, child ULs should have a 4 space indent.

... nope

Because `0` is falsy, it ignores the `start_indent` value and instead, enforces the `start_indented` as `true` (not what I want) and defaults to the `indent` value (4 spaces). If I change the `start_indent` to `3` or `-1` or anything other than `0`, it will actually enforce the starting indentation, but it ignores the only value I want, `0`.

I JUST WANT NONE! Give me back this functionality that was deprecated in 0.19.0, and removed in 0.32.0. I'm pretty sure it was removed on accident while trying to merge things with the backwards version of MD007.

Honestly, the `start_indented` param should just be removed and you should use `indent` for your indentation, and `start_indent` for the first level indent, which should default to `0`. This completely removes the need for the very confusing, and seemingly broken `start_indented`.

**Comment by DavidAnson** [comment]
Indenting content by 4 spaces creates an indented code block per the CommonMark specification: <https://spec.commonmark.org/0.31.2/#indented-code-blocks>.

As such, the example above is not an indented list and therefore rules about indented lists do not apply.

That said, the observation about incorrect handling of Number rule parameters with value 0 is valid and has been addressed by <https://github.com/DavidAnson/markdownlint/commit/e244407a664b5bb1b2eaf83aee9daf56512b73f6>.

**Comment by TheJaredWilcurt** [comment]
Thank you! And the four-space code-fence makes sense.

When will there be a new release with this change?

**Comment by DavidAnson** [comment]
I tend to release every few months. Historically: https://www.npmjs.com/package/markdownlint?activeTab=versions

**Comment by TheJaredWilcurt** [comment]
<img width="160" height="160" alt="Image" src="https://github.com/user-attachments/assets/c950dc4b-a99c-4ed0-8320-bfed21e9a15a" />

**Comment by DavidAnson** [comment]
I'll probably do a release in the next week or so.

**Comment by TheJaredWilcurt** [comment]
<img width="220" height="198" alt="thank you" src="https://github.com/user-attachments/assets/e657c8aa-ee59-49e3-98b1-b83696036a16" />
